src/tools/dart-run.md

```diff
@@ -0,0 +1,74 @@
+---
+title: dart run
+description: Use dart run to run a Dart script in your package.
+---
+
+The `dart run` command supports running a Dart script located in a file,
+or a package, or in one of its dependencies, from the command line.
+
+To run an executable when you are not currently inside a package,
+use the [pub global](/tools/pub/cmd/pub-global) command.
+
+The `dart run` command combines what was previously exposed as `pub run`
+and as the [Dart VM](/tools/dart-vm).
+
+{% include tools/dart-tool-note.md %}
+
+Here's an example of creating a new app and running it:
+
+```terminal
+$ dart create myapp
+$ cd myapp
+$ dart run
+```
+
+## Running a script in your package's bin directory
+
+This is the simplest use case.
+
+To run a script in a package named `foo`,
+which contains `bar.dart` in the `bin` directory,
+run the following command from the root of the package,
+optionally passing zero or more arguments:
+
+```terminal
+$ dart run foo:bar [arg1 arg2 ...]
+```
+
+If your package and script file in `bin` have the same name
+(e.g. `/foo/bin/foo.dart`),
+then you can use the following shorthand:
+
+```terminal
+$ dart run foo [arg1 arg2]
+```
+
+## Running a script in another directory in your package
+
+To run a script inside a directory other than the top-level
+bin directory (but within the package),
+prepend the relative path to the name of the script.
+
+For example, to run `foo.dart` in the `example/sub` directory:
+
+```terminal
+$ dart run example/sub/foo.dart [arg1 arg2 ...]
+```
+
+## Running a script in a dependency
+
+To run a script from the `bin` directory of a package that you depend on
+in the pubspec, specify the package name.
+For example, to run `bar.dart` in the `baz` package:
+
+```terminal
+$ dart run baz:bar [arg1 arg2 ...]
```

Comment:
Here it might be nice to illustrate the shorthand again... This is usually where it's useful.

`dart test` is actually a shorthand for `dart run test` which is a shorthand for `dart run test:test`.

And this is actually the case where you're running a script in a dependency.